jenkins/jobs/bubble_jobs.groovy

```diff
@@ -0,0 +1,135 @@
+def ORGANIZATION_UTILS_GITHUB_DEFAULT_BRANCH = 'master'
+
+def BUBBLE_BLUEPRINT_NAME                    = 'bubble-blueprint'
+def BUBBLE_BLUEPRINT_GITHUB_REPOSITORY       = "${ORGANIZATION_NAME}/${BUBBLE_BLUEPRINT_NAME}"
+def BUBBLE_BLUEPRINT_GITHUB_DEFAULT_BRANCH   = 'master'
+
+def BUBBLE_COOKBOOK_NAME                     = 'bubble'
+
+def MCCD_JENKINS_GITHUB_CREDENTIALS       = 'f4ec9d6e-49fb-497c-bd1f-e42d88e105da'
+
+def DEFAULT_EXECUTOR = 'executor'
+
+folder(FOLDER_NAME)
+
+freeStyleJob(SEED_JOB) {
+  logRotator {
+    numToKeep(10)
+    artifactNumToKeep(10)
+  }
+  label(DEFAULT_EXECUTOR)
+  scm {
+    git {
+      remote {
+        github(ORGANIZATION_UTILS_GITHUB_REPOSITORY, 'https' )
+      }
+      branch(ORGANIZATION_UTILS_GITHUB_DEFAULT_BRANCH)
+      shallowClone(true)
+      clean(true)
+      configure { node ->
+        node / 'extensions' << 'hudson.plugins.git.extensions.impl.PathRestriction' {
+          includedRegions 'jenkins/jobs/bubble_jobs[.]groovy'
+          excludedRegions ''
+        }
+      }
+    }
+  }
+  triggers {
+    githubPush()
+  }
+  steps {
+    dsl {
+      external('jenkins/jobs/bubble_jobs.groovy')
+    }
+  }
+}
+
+freeStyleJob(TRACKING_REPO_UPDATE_JOB) {
+  displayName('bubble-blueprint tracking repo update')
+  parameters {
+    stringParam(DEFAULT_GIT_REPO_BRANCH_PARAM, BUBBLE_BLUEPRINT_GITHUB_DEFAULT_BRANCH, 'Branch to be built')
+  }
+  label(DEFAULT_EXECUTOR)
+  concurrentBuild()
+  throttleConcurrentBuilds {
+    maxPerNode(1)
+  }
+  logRotator {
+    numToKeep(50)
+    artifactNumToKeep(10)
+  }
+  wrappers {
+    colorizeOutput('xterm')
+    timestamps()
+  }
+  triggers {
+    cron('H/15 * * * *')
+  }
+  scm {
+    git {
+      remote {
+        github(BUBBLE_BLUEPRINT_GITHUB_REPOSITORY, 'ssh' )
+        credentials(MCCD_JENKINS_GITHUB_CREDENTIALS)
+      }
+      branch(injectJobVariable(DEFAULT_GIT_REPO_BRANCH_PARAM))
+      shallowClone(false)
+      clean(true)
+      recursiveSubmodules(true)
+      trackingSubmodules(true)
+    }
+  }
+  steps {
+    shell(makeMultiline([
+      "git config --global user.email \"int-mccd_jenkins@schubergphilis.com\"",
+      "git config --global user.name \"mccd-jenkins\"",
+      "if [ -z \"\$(git status -su)\" ]; then",
+      "  echo \"==> No submodule changed\"",
+      "else",
+      "  echo \"==> Submodule has changed in remote repository\"",
+      "",
+      "  echo \"=> Clean outdated cookbooks\"",
+      "  find chef-repo/cookbooks/* -maxdepth 0 -type d | grep -v '${BUBBLE_COOKBOOK_NAME}' | xargs rm -rf",
+      "",
+      "  echo \"=> Clean berkshelf cache\"",
+      "  rm -rf ~/.berkshelf/cookbooks/*",
+      "",
+      "  TMP_UUID=\"/tmp/`uuidgen`\"",
+      "  echo \"=> Create tmp working directory \$TMP_UUID\"",
+      "  mkdir \$TMP_UUID",
+      "",
+      "  echo \"=> Berks vendor the cookbook\"",
+      "  berks vendor --berksfile=chef-repo/cookbooks/${BUBBLE_COOKBOOK_NAME}/Berksfile \$TMP_UUID",
```

Comment:
Do you want `${BUBBLE_COOKBOOK_NAME}` to be expanded when creating the job or when executing the job?

It seems to me that, I this line it will be expanded when you create the job, while in Line 114 it will be expanded when executing the job.